### ISSUE TITLE
Split Validatable in EmailValidatable and PasswordValidatable

### ISSUE THREAD

**Issue by rhymes** [plataformatec/devise#4190]
My use case is a single user table, each user belongs to an application, so the emails have to be unique within that application id, not globally. See https://github.com/plataformatec/devise/blob/master/lib/devise/models/validatable.rb#L30

Hence I cannot use Validatable as is because it does not allow a scoping. The password bit instead it's perfect as-is.

My proposal is to discuss the possibility to split the module in two separate modules (keeping the current Validatable as-is so it won't break existing code). 

Basically Validatable will include EmailValidatable and PasswordValidatable so end-users of Devise can either include both or one of them.

The alternative is two make those validators customizable (with options passed somehow) but it seems to be trickier.

What do you think?

**Comment by lucasmazza** [comment]
@rhymes I would recommend rolling your own validations instead of we open room for customization on the `Validatable` module right now - for a scenario such as yours makes sense to go for app specific validations even if it needs to duplicate some of the 6 validations we have builtin.

**Comment by rhymes** [comment]
@lucasmazza yeah, that's what I did. I just thought it made sense to split the module in two and put Validatable as-is as a facade on those two modules. Thanks anyway!
